### ISSUE TITLE
8.2.0 extract breaks with --null-as-default-value

### ISSUE THREAD

**Issue by emmaengl** [vendure-ecommerce/ngx-translate-extract#18]
Since updating to `8.2.0` the `ngx-translate-extract` command fails with following error when using the `--null-as-default-value` flag:

`An error occurred: TypeError: Cannot read properties of null (reading 'value')`

No issues with parsing pre-existing keys, only on adding new ones. The issue persists on `8.2.1`, my guess is that it's related to the type checking introduced in `8.2.0`. Can someone take a look? 🙏

**Comment by Wradgio** [comment]
I have the same issue. Created a new project with only one file as a source of translation strings. Running `ngx-translate-extract --input ./projects/xxx --output ./projects/xxx/i18n/{de,fr}.json --sort --clean --format 'namespaced-json' -n --format-indentation '  '"` returns:

`An error occurred: TypeError: Cannot read properties of null (reading 'value')`

It works after running without `-n` resp.  `--null-as-default-value` flag, but of course, strings are empty by default.

**Comment by michaelbromley** [comment]
Hi,
thanks for the report. Can you confirm that the last working version was?

**Comment by emmaengl** [comment]
@michaelbromley for us it was `8.1.1`

**Comment by michaelbromley** [comment]
Paging @SamanthaAdrichem - off the top of your head can you think of anything in your work that might have affected this `--null-as-default-value` flag?

**Comment by SamanthaAdrichem** [comment]
No, but its most likely caused by not creating the new interface on the default null values but just null.

it's probably something like
```
    key: null
```

instead of the new
```
   key: <NewInterface>{value: null, sources: []}
```

(didn't look at the code, done this from memory so will have many typo's)

I'll push a fix this afternoon

**Comment by SamanthaAdrichem** [comment]
See PR :)

**Comment by emmaengl** [comment]
@SamanthaAdrichem , thanks for getting to this so quickly, can confirm it works for me 🎉

**Comment by michaelbromley** [comment]
Thanks all for your quick attention! I just released v8.2.2 which contains this fix 👍
